src/google/protobuf/compiler/cpp/cpp_helpers.h

```diff
@@ -74,6 +74,11 @@ string SuperClassName(const Descriptor* descriptor);
 // anyway, so normally this just returns field->name().
 string FieldName(const FieldDescriptor* field);
 
+// Get the sanitized name that should be used for the  given enum in C++ code.
+// The name is optionally mangled for reserved keyword collisions. (Nested
+// names need not be mangled as they receive a prefix from the outer message type.)
+string EnumName(const EnumValueDescriptor* enum_value, bool mangle);
```

Comment:
Yes, the recommended way is to reference the nested enum from class scope,
rather then using the underscore separated names. Also all the values
should be capitalized as suggested by the style guide. The conflicts should
not be common.

On Mon, Mar 16, 2015 at 10:58 AM cconroy <notifications@github.com> wrote:

> In src/google/protobuf/compiler/cpp/cpp_helpers.h
> <https://github.com/google/protobuf/pull/228#discussion_r26510550>:
>
> > @@ -74,6 +74,11 @@ string SuperClassName(const Descriptor* descriptor);
> >  // anyway, so normally this just returns field->name().
> >  string FieldName(const FieldDescriptor* field);
> >
> > +// Get the sanitized name that should be used for the  given enum in C++ code.
> > +// The name is optionally mangled for reserved keyword collisions. (Nested
> > +// names need not be mangled as they receive a prefix from the outer message type.)
> > +string EnumName(const EnumValueDescriptor* enum_value, bool mangle);
>
> Are you sure you want it to always be mangled?
>
> That means we go from e.g. TestConflictingEnumNames_
> NestedConflictingEnum_and to TestConflictingEnumNames_
> NestedConflictingEnum_and_ ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/protobuf/pull/228/files#r26510550>.
>